sources/web/datalab/static/page.js

```diff
@@ -0,0 +1,231 @@
+function placeHolder() {}
+
+function reportEvent(event) {
+  var reportingEnabled = (document.body.getAttribute('data-reporting-enabled') == 'true');
+  if (!reportingEnabled) { return; }
+
+  var signedIn = (document.body.getAttribute('data-signed-in') == 'true');
+  var additionalMetadata = 'signedIn=' + signedIn;
+  if (event['metadata']) {
+    event['metadata'] = event['metadata'] + ',' + additionalMetadata;
+  } else {
+    event['metadata'] = additionalMetadata;
+  }
+  window.dataLayer = window.dataLayer || [];
+  window.dataLayer.push(event);
+}
+
+function toggleSidebar() {
```

Comment:
I think `toggleSidebar`, `showHelp` (and the following markup snippet), and lines 197-227 should move into notebook-app.js.

Should this file be called app-bar.js? It consists mostly of the common app bar scripts, like the About, restarting and managing VM, sign in/out.. etc. Common code like the xhr function should perhaps stay in datalab.js.